src/Core/View.js

```diff
@@ -291,8 +295,31 @@ View.prototype.addLayer = function addLayer(layer, parentLayer) {
 View.prototype.notifyChange = function notifyChange(needsRedraw, changeSource) {
     this._changeSources.add(changeSource);
     this.mainLoop.scheduleViewUpdate(this, needsRedraw);
+    if (this.avoidCollision && (changeSource === undefined || changeSource.type != 'Mesh')) {
+        this.checkCollisionWithGeometryLayer();
+    }
+};
+
+
+/**
+ * Test for collision between camera and geometry layer (DTM/DSM) to adjust camera position
+ */
+View.prototype.checkCollisionWithGeometryLayer = function checkCollisionWithGeometryLayer() {
+    const camLocation = new Coordinates(this.referenceCrs, this.camera.camera3D.position).as('EPSG:4326');
+    if (this._layers[0] && this._layers[0] instanceof GeometryLayer) {
```

Comment:
You can't really expect the first layer to always be the geometry one (even if it's true in GlobeView/PlanarView)